src/write-archive/archive-file.ts

```diff
@@ -0,0 +1,88 @@
+import mime from 'mime';
+import path from 'path';
+import { createHash } from 'node:crypto';
+import { logger } from '../utils/logger';
+import type { ArchiveResponse, UrlString } from '../resource-archive';
+
+export class ArchiveFile {
+  url: URL;
+
+  response: ArchiveResponse;
+
+  shortenedFileNameLength: number;
+
+  constructor(url: UrlString, response: ArchiveResponse) {
+    this.url = new URL(url);
+    this.response = response;
+    this.shortenedFileNameLength = 250;
+  }
+
+  originalSrc() {
+    // Assets that we capture will be stripped of the domain in the source
+    return `${this.url.pathname}${this.url.search}`;
```

Comment:
Technically this will also strip out the URL hash. If we're dealing with HTML URLs here that could matter; if we're not, it shouldn't affect things.

Reply:
Ya good call! I left those off because they don't get sent to the server so shouldn't affect the response body like a query string can.